app/code/Magento/Deploy/Model/Filesystem.php

```diff
@@ -152,19 +153,35 @@ public function __construct(
      * @return void
      * @throws LocalizedException
      * @throws \Exception
+     * @deprecated
+     * @see regenerateStaticFiles
      */
-    public function regenerateStatic(
-        OutputInterface $output
-    ) {
+    public function regenerateStatic(OutputInterface $output)
+    {
+        $this->regenerateStaticFiles($output, false);
+    }
+
+    /**
+     * Regenerate static
+     *
+     * @param OutputInterface $output
+     * @param bool $resetStaticVersion
+     * @return void
+     * @throws LocalizedException
+     * @throws \Exception
+     */
+    public function regenerateStaticFiles(OutputInterface $output, $resetStaticVersion = false)
```

Comment:
Unfortunately, we are not allowed to add new public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)